### PR TITLE
[Feature][Connector][Http] Support multi-line text splits

### PR DESCRIPTION
**Pull request by HsbcJone**
https://github.com/apache/incubator-seatunnel/issues/4649

## Purpose of this pull request
[Feature][Connector][Http] Support multi-line text splits 

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [x] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/incubator-seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by HsbcJone** [comment]
The code has been submitted and local testing has passed. We hope you can review it

File: seatunnel-connectors-v2/connector-http/connector-http-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/http/config/HttpConfig.java [hunk 1]
```diff
@@ -83,6 +84,13 @@ public class HttpConfig {
                     .withDescription(
                             "SeaTunnel content field.This parameter can get some json data, and there is no need to configure each field separately.");
 
+    public static final Option<Boolean> ENABLEMULTILINES =
+            Options.key("enableMultiLines")
```
Comment (hailin0) [inline seatunnel-connectors-v2/connector-http/connector-http-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/http/config/HttpConfig.java]:
```suggestion
            Options.key("enable_multi_lines")
```

File: seatunnel-connectors-v2/connector-http/connector-http-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/http/config/HttpConfig.java [hunk 2]
```diff
@@ -83,6 +84,13 @@ public class HttpConfig {
                     .withDescription(
                             "SeaTunnel content field.This parameter can get some json data, and there is no need to configure each field separately.");
 
+    public static final Option<Boolean> ENABLEMULTILINES =
```
Comment (hailin0) [inline seatunnel-connectors-v2/connector-http/connector-http-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/http/config/HttpConfig.java]:
```suggestion
    public static final Option<Boolean> ENABLE_MULTI_LINES =
```

File: seatunnel-connectors-v2/connector-http/connector-http-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/http/source/HttpSourceReader.java [hunk 3]
```diff
@@ -130,6 +132,24 @@ public void pollNext(Collector<SeaTunnelRow> output) throws Exception {
...
     }
 
+    /**
+     * deal lineData
+     *
+     * @param output
+     * @param data
+     * @throws IOException
+     */
+    private void dealLineData(Collector<SeaTunnelRow> output, String data) throws IOException {
```
Comment (hailin0) [inline seatunnel-connectors-v2/connector-http/connector-http-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/http/source/HttpSourceReader.java]:
```suggestion
    private void collect(Collector<SeaTunnelRow> output, String data) throws IOException {
```

File: seatunnel-connectors-v2/connector-http/connector-http-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/http/source/HttpSourceReader.java [hunk 4]
```diff
@@ -130,6 +132,24 @@ public void pollNext(Collector<SeaTunnelRow> output) throws Exception {
         }
     }
 
+    /**
+     * deal lineData
+     *
+     * @param output
+     * @param data
+     * @throws IOException
+     */
```
Comment (hailin0) [inline seatunnel-connectors-v2/connector-http/connector-http-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/http/source/HttpSourceReader.java]:
```suggestion
```

File: docs/en/connector-v2/source/Http.md [hunk 5]
```diff
@@ -17,23 +17,24 @@ Used to read data from Http.
...
+| method                      | String  | No       | get           |
+| headers                     | Map     | No       | -             |
+| params                      | Map     | No       | -             |
+| body                        | String  | No       | -             |
+| poll_interval_ms            | int     | No       | -             |
+| retry                       | int     | No       | -             |
+| retry_backoff_multiplier_ms | int     | No       | 100           |
+| retry_backoff_max_ms        | int     | No       | 10000         |
+| common-options              |         | No       | -             |
+| enableMultiLines            | boolean | No       | false         |
```
Comment (hailin0) [inline docs/en/connector-v2/source/Http.md]: Is this setting required for connectors other than http?

**Comment by hailin0** [comment]
add e2e testcase into this case

https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/connector-http-e2e/src/test/java/org/apache/seatunnel/e2e/connector/http/HttpIT.java#L91

**Comment by TyrantLucifer** [comment]
> add e2e testcase into this case
> 
> https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/connector-http-e2e/src/test/java/org/apache/seatunnel/e2e/connector/http/HttpIT.java#L91

+1

File: docs/en/connector-v2/source/Http.md (hunk 5 above)
Comment (HsbcJone) [reply docs/en/connector-v2/source/Http.md]: > Is this setting required for connectors other than http?

Yes, I think this configuration is universal.

**Comment by HsbcJone** [comment]
> > add e2e testcase into this case
> > https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/connector-http-e2e/src/test/java/org/apache/seatunnel/e2e/connector/http/HttpIT.java#L91
> 
> +1

get

**Comment by hailin0** [comment]
check ci error

**Comment by HsbcJone** [comment]
> check ci error

done

**Comment by HsbcJone** [comment]
I have added **e2e testing classes** and optimized the code style . @hailin0 @TyrantLucifer

File: docs/en/connector-v2/source/Github.md [hunk 6]
```diff
@@ -17,22 +17,23 @@ Used to read data from Github.
...
+| params                      | Map     | No       | -             |
+| body                        | String  | No       | -             |
+| json_field                  | Config  | No       | -             |
+| content_json                | String  | No       | -             |
+| poll_interval_ms            | int     | No       | -             |
+| retry                       | int     | No       | -             |
+| retry_backoff_multiplier_ms | int     | No       | 100           |
+| retry_backoff_max_ms        | int     | No       | 10000         |
+| common-options              | config  | No       | -             |
+| enableMultiLines            | boolean | No       | false         |
```
Comment (hailin0) [inline docs/en/connector-v2/source/Github.md:36]: Update all this configuration item name

Comment (hailin0) [inline docs/en/connector-v2/source/Github.md]:
```suggestion
| enable_multi_lines           | boolean | No       | false         |
| common-options              | config  | No       | -             |
```

Comment (hailin0) [reply docs/en/connector-v2/source/Github.md]: Update all this configuration item name

**Comment by hailin0** [comment]
check ci error

File: docs/en/connector-v2/source/Github.md (hunk 6 above)
Comment (HsbcJone) [reply docs/en/connector-v2/source/Github.md]: done

File: docs/en/connector-v2/source/Github.md [hunk 7]
```diff
@@ -17,22 +17,23 @@ Used to read data from Github.
...
+| params                      | Map     | No       | -             |
+| body                        | String  | No       | -             |
+| json_field                  | Config  | No       | -             |
+| content_json                | String  | No       | -             |
+| poll_interval_ms            | int     | No       | -             |
+| retry                       | int     | No       | -             |
+| retry_backoff_multiplier_ms | int     | No       | 100           |
+| retry_backoff_max_ms        | int     | No       | 10000         |
+| common-options              | config  | No       | -             |
+| enable_multi_lines            | boolean | No       | false         |
```
Comment (hailin0) [inline docs/en/connector-v2/source/Github.md]: move to the front of `common-options`

Comment (hailin0) [reply docs/en/connector-v2/source/Github.md]: update all this config item

Comment (HsbcJone) [reply docs/en/connector-v2/source/Github.md]: done

**Comment by hailin0** [comment]
image

**Comment by HsbcJone** [comment]
> image

Fixed this issue

**Comment by HsbcJone** [comment]
> > image
> 
> Fixed this issue

CI-All checks have passed

**Review by hailin0 — APPROVED** [review]

**Review by davidzollo — APPROVED** [review]
+1
LGTM

**Pull request merged**